components/script/script_thread.rs

```diff
@@ -1471,28 +1476,292 @@ impl ScriptThread {
         debug!("Stopped script thread.");
     }
 
-    /// Handle incoming control messages.
-    fn handle_msgs(&self) -> bool {
-        use self::MixedMessage::{
-            FromConstellation, FromDevtools, FromImageCache, FromScript, FromWebGPUServer,
+    /// <https://drafts.csswg.org/cssom-view/#document-run-the-resize-steps>
+    fn run_the_resize_steps(
+        &self,
+        id: PipelineId,
+        size: WindowSizeData,
+        size_type: WindowSizeType,
+    ) {
+        self.profile_event(ScriptThreadEventCategory::Resize, Some(id), || {
+            self.handle_resize_event(id, size, size_type);
+        });
+    }
+
+    /// Process compositor events as part of a "update the rendering task".
+    fn process_pending_compositor_events(&self, pipeline_id: PipelineId) {
+        let document = match self.documents.borrow().find_document(pipeline_id) {
+            Some(document) => document,
+            None => {
+                return warn!(
+                    "Processing pending compositor events for closed pipeline {}.",
+                    pipeline_id
+                )
+            },
         };
+        // Do not handle events if the BC has been, or is being, discarded
+        if document.window().Closed() {
+            return warn!(
+                "Compositor event sent to a pipeline with a closed window {}.",
+                pipeline_id
+            );
+        }
+        let window = document.window();
+        ScriptThread::set_user_interacting(true);
+        let _realm = enter_realm(document.window());
+        for event in document.get_pending_compositor_events().into_iter() {
```

Comment:
```suggestion
        for event in document.get_pending_compositor_events().into_iter() {
```

If you are just calling `into_iter()` on this then maybe `Vec` is a better choice than `VecDeque`?